src/components/Wallet/WalletSign.jsx

```diff
@@ -30,68 +29,27 @@ class WalletSign extends React.Component {
     };
   }
 
+  static getDerivedStateFromProps(props, state) {
+    const { txid, changeSlice, updateTxSlices } = props;
+    if (txid.length && !state.spent) {
+      updateTxSlices(changeSlice);
+      return {
+        spent: true,
+      };
+    }
+    return null;
+  }
+
   componentWillUnmount() {
-    const {
-      resetTransaction,
-      transaction,
-      changeSlice,
-      spendSlices,
-    } = this.props;
+    const { resetTransaction } = this.props;
     const { spent } = this.state;
+
+    // reset the transaction when we leave the view if tx is spent
     if (spent) {
-      // have a brief delay to make sure the queried node had enough time
-      // to update
-      setTimeout(() => spendSlices(transaction.inputs, changeSlice), 1000);
       resetTransaction();
     }
   }
 
-  render = () => {
```

Comment:
This just got moved to be the last function which is what we do in most of our other components.